pkg/standalone-utils/contracts/AumProtocolFeesCollector.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/InputHelpers.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/Authentication.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/ReentrancyGuard.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/SafeERC20.sol";
+
+import "./interfaces/IAumProtocolFeesCollector.sol";
+
+/**
+ * @dev This an auxiliary contract to the Vault, deployed by it during construction. It offloads some of the tasks the
+ * Vault performs to reduce its overall bytecode size.
+ *
+ * The current values for all protocol fee percentages are stored here, and any tokens charged as protocol fees are
+ * sent to this contract, where they may be withdrawn by authorized entities. All authorization tasks are delegated
+ * to the Vault's own authorizer.
+ */
+contract AumProtocolFeesCollector is IAumProtocolFeesCollector, Authentication, ReentrancyGuard {
```

Comment:
I'd be inclined to have this be a simple contract which only returns the protocol AUM fee percentage and rely on the standard protocol fee collector to custody the funds.

Reply:
I still wanted it to be controlled by governance, but the consensus seems to be send all funds to the Vault's fee collector, so this version does that. (See the updated comment.)